### PR TITLE
chore(scanner): followup to #6663

### PR DESCRIPTION
**Pull request by RTann**
## Description

Addressing some leftover comments: https://github.com/stackrox/stackrox/pull/6663

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

No

### CODE REVIEW

File: scanner/cmd/scanner/main.go [hunk 1]
```diff
@@ -72,30 +71,14 @@ func main() {
 
 	// Wait for signals.
 	sigC := make(chan os.Signal, 1)
-	signal.Notify(sigC, syscall.SIGINT, syscall.SIGTERM)
+	signal.Notify(sigC, os.Interrupt, unix.SIGTERM)
 	sig := <-sigC
-	zlog.Warn(ctx).Str("signal", sig.String()).Send()
+	zlog.Info(ctx).Str("signal", sig.String()).Send()
```
Comment (jvdm) [inline scanner/cmd/scanner/main.go:76]: `SIGTERM` is expected to happen in a pod's lifecycle, care to clarify why the change?

File: scanner/cmd/scanner/main.go [hunk 2]
```diff
@@ -72,30 +71,14 @@ func main() {
 
 	// Wait for signals.
 	sigC := make(chan os.Signal, 1)
-	signal.Notify(sigC, syscall.SIGINT, syscall.SIGTERM)
+	signal.Notify(sigC, os.Interrupt, unix.SIGTERM)
```
Comment (jvdm) [inline scanner/cmd/scanner/main.go:74]: Care to clarify why?

File: scanner/cmd/scanner/logging.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+		return err
+	}
+	logger := zerolog.New(os.Stdout).
+		Level(zerolog.DebugLevel).
+		With().
+		Timestamp().
+		Str("host", hostname).
+		Logger()
+	zlog.Set(&logger)
+	log.Logger = zerolog.Nop()
```
Comment (jvdm) [inline scanner/cmd/scanner/logging.go:24]: Why is this needed?

File: scanner/cmd/scanner/logging.go
Comment (jvdm) [inline scanner/cmd/scanner/logging.go:1]: I believe that dedicating an entire file solely to an initialization function can potentially complicate code necessarily.

File: scanner/cmd/scanner/main.go [hunk 4]
```diff
@@ -160,16 +143,16 @@ func createBackends(ctx context.Context) (*Backends, error) {
 }
 
 func (b *Backends) Close(ctx context.Context) {
-	if b.Matcher != nil {
-		err := b.Matcher.Close(ctx)
-		if err != nil {
-			zlog.Error(ctx).Err(err).Msg("closing matcher")
-		}
-	}
```
Comment (jvdm) [inline scanner/cmd/scanner/main.go:168]: Matcher was initialized last, so it is closed first. Why are you changing the order?

File: scanner/cmd/scanner/logging.go (hunk 3 above)
Comment (jvdm) [reply scanner/cmd/scanner/logging.go:24]: I saw Hank's comment. It worth adding a comment here:

```suggestion
	// Disable the default zerolog logger.
	log.Logger = zerolog.Nop()
```

**Review by jvdm — COMMENTED** [review]
I have some questions. Could you provide additional details about the changes you proposing? This would help determine whether to move forward or not. Thank you very much.

File: scanner/cmd/scanner/logging.go
Comment (RTann) [reply scanner/cmd/scanner/logging.go:1]: yeah I don't like having the extra file. I originally did it because we would have had multiple `log` packages, and I didn't like any of the options I came up with for renaming `github.com/rs/zerolog/log`. However, I realize I can just resolve this by renaming `log` as `golog` or something

File: scanner/cmd/scanner/main.go (hunk 2 above)
Comment (RTann) [reply scanner/cmd/scanner/main.go:74]: copying over from https://github.com/stackrox/stackrox/pull/6663#discussion_r1266040876:

> something I forgot: I don't think syscall is really supposed to be used directly anymore: https://pkg.go.dev/syscall#pkg-overview
>
> The current Scanner v2 and Sensor both watch os.Interrupt, unix.SIGTERM - stackrox/rox#5024

File: scanner/cmd/scanner/main.go (hunk 1 above)
Comment (RTann) [reply scanner/cmd/scanner/main.go:76]: if it's expected, then doesn't `info` make more sense than `warn`? `warn` sounds like it makes more sense for unexpected, yet ok, things

File: scanner/cmd/scanner/main.go (hunk 4 above)
Comment (RTann) [reply scanner/cmd/scanner/main.go:168]: asked about it in the previous PR: https://github.com/stackrox/stackrox/pull/6663#discussion_r1266043358

Just wanted to understand why it was in that order in the first place. Makes sense now. I will revert this

File: scanner/cmd/scanner/main.go (hunk 2 above)
Comment (jvdm) [reply scanner/cmd/scanner/main.go:77]: I see, let's do this then:

```
signal.Notify(sigC, unix.SIGINT, unix.SIGTERM)
```

File: scanner/cmd/scanner/main.go [hunk 5]
```diff
@@ -4,14 +4,14 @@ import (
...
-	"log"
+	golog "log"
 	"os"
 	"os/signal"
 	"path/filepath"
-	"syscall"
 
 	"github.com/quay/zlog"
 	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
```
Comment (jvdm) [inline scanner/cmd/scanner/main.go:14]: Nitpick: It would make more sense to rename the zero log package to `zlog` or something since someone reading the code would assume `log` means the default package `"log"`.

**Review by jvdm — COMMENTED** [review]
Almost there, just a nitpick that you can ignore and ask to use `unix` in signals.

**Comment by RTann** [comment]
When migrating the Scanner module into `rox`, I ran into this:

```
$ make style-slim
+ roxvet
+ roxvet
# github.com/tecbot/gorocksdb
../../../../pkg/mod/github.com/!data!dog/gorocksdb@v0.0.0-20200107201226-9722c3a2e063/array.go:4:11: fatal error: 'rocksdb/c.h' file not found
 #include "rocksdb/c.h"
          ^~~~~~~~~~~~~
1 error generated.
# github.com/stackrox/rox/scanner/cmd/scannercli
scanner/cmd/scannercli/main.go:24:12: error result of func Setenv is not being checked
scanner/cmd/scannercli/main.go:25:12: error result of func Setenv is not being checked
scanner/cmd/scannercli/main.go:26:12: error result of func Setenv is not being checked
scanner/cmd/scannercli/main.go:27:12: error result of func Setenv is not being checked
scanner/cmd/scannercli/main.go:49:18: error result of func Close is not being checked
# github.com/stackrox/rox/scanner/indexer
scanner/indexer/service.go:27:14: Varargs function "google.golang.org/grpc/status".Errorf called with no variadic arguments; use Error instead
scanner/indexer/service.go:30:14: Varargs function "google.golang.org/grpc/status".Errorf called with no variadic arguments; use Error instead
scanner/indexer/service.go:33:14: Varargs function "google.golang.org/grpc/status".Errorf called with no variadic arguments; use Error instead
# github.com/stackrox/rox/scanner/matcher
scanner/matcher/service.go:27:14: Varargs function "google.golang.org/grpc/status".Errorf called with no variadic arguments; use Error instead
scanner/matcher/service.go:31:14: Varargs function "google.golang.org/grpc/status".Errorf called with no variadic arguments; use Error instead
# github.com/stackrox/rox/scanner/cmd/scanner
scanner/cmd/scanner/main.go:38:12: error result of func Setenv is not being checked
scanner/cmd/scanner/main.go:39:12: error result of func Setenv is not being checked
scanner/cmd/scanner/main.go:40:12: error result of func Setenv is not being checked
scanner/cmd/scanner/main.go:41:12: error result of func Setenv is not being checked
scanner/cmd/scanner/main.go:15:2: invalid import "github.com/stackrox/rox/central/grpc/metrics": scanner cannot import from central/grpc/metrics; only allowed roots are [scanner generated image pkg]
make: *** [roxvet] Error 1
```

I'm going to make these changes here

**Comment by RTann** [comment]
```
scanner/cmd/scanner/main.go:15:2: invalid import "github.com/stackrox/rox/central/grpc/metrics": scanner cannot import from central/grpc/metrics; only allowed roots are [scanner generated image pkg]
```

will be resolved in https://github.com/stackrox/stackrox/pull/7070

**Comment by RTann** [comment]
> ```
> scanner/cmd/scanner/main.go:15:2: invalid import "github.com/stackrox/rox/central/grpc/metrics": scanner cannot import from central/grpc/metrics; only allowed roots are [scanner generated image pkg]
> ```
> 
> will be resolved in #7070

I lied going to do it here

**Review by jvdm — APPROVED** [review]
LGTM.
